Changes location of MooseDocs markdown

aeslaughter opened the pull request:
This will likely require some integration with applications.
(closes #11730)

<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

aeslaughter committed:
Fix case of CentroidMultiApp

(refs #11730)

aeslaughter committed:
Add filename for register call

(refs #11730)

aeslaughter committed:
Move regeister macro to source files

(refs #11730)

aeslaughter committed:
Re-factor MooseDocs to mimic include directories for markdown locations

(closes #11730)

aeslaughter committed:
Re-organize pages for future changes to file locations

(refs #11730)

aeslaughter committed:
Move markdown to new locations

(closes #11730)

aeslaughter committed:
Update markdown content for change to file locations

(refs #11730)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/206268/) on 259821f wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11772/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11772/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 99ae987fbf3801a5d3808f3ffd0adc506b8b1224`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/206280/) on fe34618 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11772/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/206376/) on 91da8ab wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11772/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11772/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format acd55146ac38cb661e939a3fa89910b7560b49d7`

aeslaughter committed:
Keep the legacy markdown locations as the default

aeslaughter committed:
Remove legacy option

All the apps that rely on MooseDocs include framework so they can not use the legacy mode anyway
(refs #11730)

aeslaughter committed:
Update locations of tensor mechanics files

aeslaughter committed:
Handle error in syntax node markdown method when running check

aeslaughter committed:
Changes in support of application documentation

(refs #11730)
